### PR TITLE
postgresql modules: use query parameters with cursor objects

### PR DESCRIPTION
**Pull request by Andersson007**
##### SUMMARY
postgresql modules: use query parameters with cursor objects

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME
```
lib/ansible/modules/database/postgresql/postgresql_table.py
lib/ansible/modules/database/postgresql/postgresql_tablespace.py
lib/ansible/modules/database/postgresql/postgresql_user.py
```

### CODE REVIEW

**Comment by Andersson007** [comment]
this is the last part of changes like these
ready_for_review

File: changelogs/fragments/65862-postgresql_modules_use_query_params_with_cursor.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+bugfixes:
+- postgresql_table - use query parameters with cursor object (https://github.com/ansible/ansible/pull/65863).
```
Comment (kostiantyn-nemchenko) [inline changelogs/fragments/65862-postgresql_modules_use_query_params_with_cursor.yml]: wrong link?

File: lib/ansible/modules/database/postgresql/postgresql_tablespace.py [hunk 2]
```diff
@@ -248,16 +248,15 @@ def get_info(self):
...
         else:
             query = ("SELECT r.rolname, t.spcoptions, %s "
                      "FROM pg_catalog.pg_tablespace AS t "
                      "JOIN pg_catalog.pg_roles AS r "
-                     "ON t.spcowner = r.oid "
-                     "WHERE t.spcname = '%s'" % (location, self.name))
+                     "ON t.spcowner = r.oid " % location)
 
-        res = exec_sql(self, query, add_to_executed=False)
+        res = exec_sql(self, query + "WHERE t.spcname = %(name)s",
```
Comment (kostiantyn-nemchenko) [inline lib/ansible/modules/database/postgresql/postgresql_tablespace.py:258]: Just curious why you're using the where condition here and not in the sql statement above?

**Review by kostiantyn-nemchenko — CHANGES_REQUESTED** [review]

File: changelogs/fragments/65862-postgresql_modules_use_query_params_with_cursor.yml (hunk 1 above)
Comment (Andersson007) [reply changelogs/fragments/65862-postgresql_modules_use_query_params_with_cursor.yml]: thanks! a typo

File: lib/ansible/modules/database/postgresql/postgresql_tablespace.py (hunk 2 above)
Comment (Andersson007) [reply lib/ansible/modules/database/postgresql/postgresql_tablespace.py:258]: as i remember, it doesn't work if you have %s and %(blabla)s in the same row in one instruction

Comment (Andersson007) [reply lib/ansible/modules/database/postgresql/postgresql_tablespace.py:258]:
```
Python 3.6.6 (default, Mar 29 2019, 00:03:27) 
[GCC 4.8.5 20150623 (Red Hat 4.8.5-36)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> s = "%s %(blabla)s" % 'first'
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: format requires a mapping
```

**Comment by Andersson007** [comment]
ready_for_review

**Comment by Andersson007** [comment]
CC @kostiantyn-nemchenko

**Review by kostiantyn-nemchenko — APPROVED** [review]

**Comment by kostiantyn-nemchenko** [comment]
shipit

**Comment by Andersson007** [comment]
@kostiantyn-nemchenko thanks! Sorry, I forgot today is Saturday

**Comment by kostiantyn-nemchenko** [comment]
@Andersson007 No problem. Thanks for the contribution!

**Pull request merged**